cmd/snap-update-ns/entry.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces/mount"
+	"github.com/snapcore/snapd/osutil"
+)
+
+// XSnapdMode returns the file mode associated with x-snapd.mode mount option.
+// If the mode is not specified explicitly then a default mode of 0755 is assumed.
+func XSnapdMode(e *mount.Entry) (os.FileMode, error) {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd.mode=") {
+			kv := strings.SplitN(opt, "=", 2)
+			var mode os.FileMode
+			n, err := fmt.Sscanf(kv[1], "%o", &mode)
```

Comment:
The Sscanf here is fine but I wonder if it would be more robust with a regex to validate the input?

Reply:
Good idea, done with 2253da23adf2469b6d3ce0135b1027c9373ff4dd